typify/tests/schemas/reflexive.rs

```diff
@@ -1,15 +1,416 @@
+    fn from(value: &CoreSchemaMetaSchemaObjectItems) -> Self {
+        value.clone()
+    }
+}
+impl Default for CoreSchemaMetaSchemaObjectItems {
+    fn default() -> Self {
+        CoreSchemaMetaSchemaObjectItems::Variant0(CoreSchemaMetaSchema::Boolean(true))
+    }
+}
+impl From<CoreSchemaMetaSchema> for CoreSchemaMetaSchemaObjectItems {
+    fn from(value: CoreSchemaMetaSchema) -> Self {
+        Self::Variant0(value)
+    }
+}
+impl From<SchemaArray> for CoreSchemaMetaSchemaObjectItems {
+    fn from(value: SchemaArray) -> Self {
+        Self::Variant1(value)
+    }
+}
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum CoreSchemaMetaSchemaObjectType {
+    Variant0(SimpleTypes),
+    Variant1(Vec<SimpleTypes>),
+}
+impl From<&CoreSchemaMetaSchemaObjectType> for CoreSchemaMetaSchemaObjectType {
+    fn from(value: &CoreSchemaMetaSchemaObjectType) -> Self {
+        value.clone()
+    }
+}
+impl From<SimpleTypes> for CoreSchemaMetaSchemaObjectType {
+    fn from(value: SimpleTypes) -> Self {
+        Self::Variant0(value)
+    }
+}
+impl From<Vec<SimpleTypes>> for CoreSchemaMetaSchemaObjectType {
+    fn from(value: Vec<SimpleTypes>) -> Self {
+        Self::Variant1(value)
+    }
+}
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NonNegativeInteger(pub u64);
+impl std::ops::Deref for NonNegativeInteger {
+    type Target = u64;
+    fn deref(&self) -> &u64 {
+        &self.0
+    }
+}
+impl From<NonNegativeInteger> for u64 {
+    fn from(value: NonNegativeInteger) -> Self {
+        value.0
+    }
+}
+impl From<&NonNegativeInteger> for NonNegativeInteger {
+    fn from(value: &NonNegativeInteger) -> Self {
+        value.clone()
+    }
+}
+impl From<u64> for NonNegativeInteger {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl std::str::FromStr for NonNegativeInteger {
+    type Err = <u64 as std::str::FromStr>::Err;
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        Ok(Self(value.parse()?))
+    }
+}
+impl std::convert::TryFrom<&str> for NonNegativeInteger {
+    type Error = <u64 as std::str::FromStr>::Err;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        value.parse()
+    }
+}
+impl std::convert::TryFrom<&String> for NonNegativeInteger {
+    type Error = <u64 as std::str::FromStr>::Err;
+    fn try_from(value: &String) -> Result<Self, Self::Error> {
+        value.parse()
+    }
+}
+impl std::convert::TryFrom<String> for NonNegativeInteger {
+    type Error = <u64 as std::str::FromStr>::Err;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        value.parse()
+    }
+}
+impl ToString for NonNegativeInteger {
+    fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+}
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NonNegativeIntegerDefault0 {
+    #[serde(flatten)]
+    pub subtype_0: NonNegativeInteger,
+    #[serde(flatten)]
+    pub subtype_1: serde_json::Value,
+}
```

Comment:
This is terrible. We need to do something significantly different to handle `allOf` constructions.